en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -0,0 +1,161 @@
+1. Execute the command `wsl --install` to run the installation routine for WSL.
+1. Reboot the system.
+1. Open `cmd` again as a normal user (not as administrator).
+   This can be done by pressing the **Start** key, typing "cmd" and pressing **Enter**.
+1. Execute the command `wsl` to access the WSL shell.
+1. WSL will prompt you for a user name and password for the Ubuntu installation.
+   **Note:** _Remember the credentials for later on!_
+
+:::note
+If you have any problems with your setup, check the current microsoft WSL installation documentation.
+:::
+
+### Install PX4 Toolchain
+
+To install the development toolchain for Gazebo simulation, JMAVSim simulation and Pixhawk/NuttX harware:
+
+1. Execute the command `cd ~` to switch to the home folder of WSL for the next steps.
+   :::warning
+   This is important!
+   If you work from a location outside of the WSL file system you'll run into issues such as very slow execution and access right/permission errors.
+   :::
+
+1. [Download PX4 Source Code](../dev_setup/building_px4.md):
+   ```bash
+   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+   ```
+1. Run the **ubuntu.sh** installer script and acknowledge any prompts as the script progresses:
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+   
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell.
+     Note that `git` is already installed in WSL.
+   
+1. Restart the "WSL computer" on completion.
+   ```
+   wsl --shutdown
+   ```
+   (or do `exit` and reopen the prompt).
+
+1. Execute command `wsl` to start WSL again and access the shell.
+1. Execute `cd ~/PX4-Autopilot` to switch to the PX4 repository in the WSL home folder.
+1. Execute `make px4_sitl` to build the PX4 SITL target and test your environment.
+
+
+### Opening a WSL shell
+
+To open the WSL shell (once WSL has been installed):
+
+1. Press the Windows **Start** key.
+1. Type "cmd" and press Enter to open a command prompt.
+1. Execute the command `wsl` to access the WSL shell.
+
+
+## Visual Studio Code Integration
+
+1. Download and install Visual Studio Code on Windows from https://code.visualstudio.com/
+2. Open _Visual Studio Code_.
+3. Install the extension called [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) (marketplace) 
+4. [Open a WSL shell](dev_env_windows_wsl.md#opening-a-wsl-shell)
+5. Switch to the PX4 folder: `cd ~/PX4-Autopilot`
+6. Open this folder with WSL integration in Visual Studio Code `code .`
+7. Now you have the full IDE integration with Terminal, make sure you always open the PX4 repository in the Remote WSL mode.
```

Comment:
What does that actually mean? I.e. does this mean I should be able to see a build target after this?

Reply:
You see everything the same as if you would be using VS code on Ubuntu with the repository opened. So you can also use the integrate terminal to execute command which is what I use in my workflow but you'll also see the configured targets and so on in the GUI.

Reply:
![image](https://user-images.githubusercontent.com/4668506/152873982-096d05ec-8512-479e-b30a-1a5b9b0aa26e.png)